packages/react/src/table/TableRow.tsx

```diff
@@ -16,9 +18,24 @@ export function TableRow({
 	'aria-rowindex': ariaRowindex,
 	children,
 	invalid,
+	onClick,
 	selected,
 }: TableRowProps) {
 	const { tableLayout } = useTableContext();
+
+	function onClickTableRow(event: React.MouseEvent<HTMLTableRowElement>) {
+		// Check when a element is positioned above the table row element
+		// Cancel the rows click event if target is not a <th>, <tr> or <td>
+		const targetElement = event.target;
+		if (!('nodeName' in targetElement)) return;
+		if (!['TH', 'TR', 'TD'].includes((targetElement as HTMLElement).nodeName))
+			return;
+
+		if (!onClick) return;
+		event.stopPropagation();
+		onClick(event);
+	}
```

Comment:
So, we'll have to change it to this. Otherwise consumers can't have spans or any other tag in their cells because the row won't click.

This code basically checks to see if what you clicked has an interactive parent and then bails. But if it then has a row as a parent, we can call the click.

```
function onClickTableRow(event: React.MouseEvent<HTMLTableRowElement>) {
		const targetElement = event.target;

		if (
			(targetElement as HTMLElement).closest(
				'a, button, input, select, textarea'
			)
		) {
			return;
		}

		const row = (targetElement as HTMLElement).closest('tr');

		if (row) {
			onClick?.(event);
		}
	}
```

Reply:
I would also rename the function to `handleOnClick`